languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/style.xml

```diff
@@ -10698,36 +10698,42 @@ USA
     </category>
 
     <category id="SCIENTIFIC" name="Linguagem científica" type="style" tone_tags="scientific">
-        <rule id="NUMERO_DE_MOLS" name="Número de mols -> massa molecular" default="temp_off" tags="picky" is_goal_specific="true">
+        <rule id="NUMERO_DE_MOLS" name="Número de mols -> massa molecular" tags="picky" is_goal_specific="true">
             <pattern>
+                <token postag_regexp="yes" postag="(SP.+:)?D.+MS.*" min="0"/>
                 <token>número</token>
                 <token>de</token>
                 <token regexp="yes">mol(e?s)?</token>
             </pattern>
-            <message>&scientific_intro; prefira <suggestion>quantidade de matéria</suggestion>.</message>
-            <example correction="quantidade de matéria">Uma amostra com alto <marker>número de mols</marker>.</example>
+            <message>&scientific_intro; prefira &quot;quantidade de matéria&quot;.</message>
+            <suggestion><match no="1" postag_regexp="yes" postag="(.+)MS(.+)" postag_replace="$1FS$2" include_skipped="all"/> quantidade de matéria</suggestion>
+            <example correction="Esta quantidade de matéria"><marker>Este número de mols</marker> da amostra.</example>
         </rule>
 
-        <rule id="PESO_MOLECULAR" name="Peso molecular -> massa molecular" default="temp_off" tags="picky" is_goal_specific="true">
+        <rule id="PESO_MOLECULAR" name="Peso molecular -> massa molecular" tags="picky" is_goal_specific="true">
             <pattern>
+                <token postag_regexp="yes" postag="(SP.+:)?D.+MS.*" min="0"/>
```

Comment:
Should we include nouns and adjectives here, to match cases like:

`Além de sua versatilidade, com a inclusão de plastificantes, o PVC vem se tornando muito aceito no mercado por possuir um material de baixo peso molecular. `

Reply:
I actually suppose we could add adjectives, not sure about nouns, though.

Reply:
No plans to work very hard on this, since you've approved I'll just merge and get back to the adjectives later.